gzip_unreliable_test.go

```diff
@@ -0,0 +1,65 @@
+// These tests are unreliable or only pass under certain conditions.
+// To run:   go test -v -count=1 -cpu=1,2,4,8,16 -tags=unreliable
+// +build unreliable,!race
+
+package pgzip
+
+import (
+	"bytes"
+	"sync"
+	"testing"
+	"time"
+)
+
+type SlowDiscard time.Duration
+
+func (delay SlowDiscard) Write(p []byte) (int, error) {
+	time.Sleep(time.Duration(delay))
+	return len(p), nil
+}
+
+// Test that the panics catch unsafe concurrent writing (a panic is better than data corruption)
+// This test is UNRELIABLE and slow. The more concurrency (GOMAXPROCS), the more likely
+// a race condition will be hit. If GOMAXPROCS=1, the condition is never hit.
+func TestConcurrentRacePanic(t *testing.T) {
+	w := NewWriter(SlowDiscard(2 * time.Millisecond))
+	w.SetConcurrency(1000, 1)
+	data := bytes.Repeat([]byte("T"), 100000) // varying block splits
+
+	const n = 1000
```

Comment:
Maybe:
```suggestion
	n := 1000
	if testing.Short() {
		n = 10
	}
```

(or make the `i++` change to bigger increments).

Though you have this guarded nicely by tags.

Reply:
This test tries to reproduce a race condition. I tried, but I cannot reproduce it with `n = 10`.

Reply:
ah, ok, no problem.